packages/back-end/src/controllers/experiments.ts

```diff
@@ -307,34 +309,57 @@ export async function getExperiment(
 
 async function _getSnapshot(
   context: ReqContext | ApiReqContext,
-  id: string,
+  experiment: string,
   phase?: string,
   dimension?: string,
   withResults: boolean = true
 ) {
-  const experiment = await getExperimentById(context, id);
+  const experimentObj = await getExperimentById(context, experiment);
 
-  if (!experiment) {
+  if (!experimentObj) {
     throw new Error("Experiment not found");
   }
 
-  if (experiment.organization !== context.org.id) {
+  if (experimentObj.organization !== context.org.id) {
     throw new Error("You do not have access to view this experiment");
   }
 
   if (!phase) {
     // get the latest phase:
-    phase = String(experiment.phases.length - 1);
+    phase = String(experimentObj.phases.length - 1);
   }
 
   return await getLatestSnapshot(
-    experiment.id,
+    experimentObj.id,
     parseInt(phase),
     dimension,
     withResults
   );
 }
 
+async function _getSnapshots(
+  context: ReqContext | ApiReqContext,
+  experiments: string[],
+  dimension?: string,
+  withResults: boolean = true
+): Promise<ExperimentSnapshotInterface[]> {
+  const experimentPhaseMap: Map<string, number> = new Map();
+  const experimentObjs = await getExperimentsByIds(context, experiments);
+  experimentObjs.forEach((e) => {
+    if (e.organization !== context.org.id) {
+      throw new Error("You do not have access to view this experiment");
+    }
+    // get the latest phase:
+    const phase = String(e.phases.length - 1);
+    experimentPhaseMap.set(e.id, parseInt(phase));
```

Comment:
Why convert length to a string and then parse it?  Can't you use the length directly?

Reply:
This was how it is done earlier in this page, but yes we should be able to. I'll make the change.